src/core/providers/accounts.ts

```diff
@@ -46,26 +44,10 @@ export function createLocalAccountsProvider(
     if (method === "eth_sendTransaction" && params.length > 0) {
       const tx: Tx = params[0];
 
-      if (obtainedChainId === undefined) {
```

Comment:
Removing this makes sense now. But we should throw an error if `tx.chainId === undefined`